task_test.go

```diff
@@ -24,12 +25,20 @@ import (
 
 func TestAllTasks(t *testing.T) {
 	client := NewFakeMarathonEndpoint(t)
-	tasks, err := client.AllTasks()
+	tasks, err := client.AllTasks(nil)
 	assert.Nil(t, err)
 	assert.NotNil(t, tasks)
 	assert.Equal(t, len(tasks.Tasks), 2)
 }
 
+func TestAllStagingTasks(t *testing.T) {
+	client := NewFakeMarathonEndpoint(t)
+	tasks, err := client.AllTasks(url.Values{"status": []string{"staging"}})
```

Comment:
I agree. As it was said let's stick with `url.Values`. We can change it in the future if we realize that we need another approach.